fvm/src/syscalls/bind.rs

```diff
@@ -94,6 +94,38 @@ fn memory_and_data<'a, K: Kernel>(
     Ok((Memory::new(mem), data))
 }
 
+fn charge_exec_units_for_gas(caller: &mut Caller<InvocationData<impl Kernel>>) -> Result<(), Trap> {
+    let exec_units = caller
+        .data_mut()
+        .calculate_exec_units_for_gas()
+        .map_err(|_| Trap::new("failed to calculate exec_units"))?;
+    if exec_units.is_negative() {
+        caller.add_fuel(u64::try_from(exec_units.saturating_neg()).unwrap_or(0))?;
+    } else {
+        caller.consume_fuel(u64::try_from(exec_units).unwrap_or(0))?;
```

Comment:
Or max, I think. We want to consume _all_ fuel if exec_units are positive and don't fit, right?

Reply:
I don't think it can not fit, right? If exec_units (and i64) is positive, it _must_ fit in a u64? We should probably just `as u64` it?